src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -266,6 +269,76 @@ class spell_sha_earth_shield : public SpellScriptLoader
         }
 };
 
+// 170374 - Earthen Rage (Passive)
+// 170377 - Earthen Rage (Proc Aura)
+class spell_sha_earthen_rage : public SpellScriptLoader
+{
+    public:
+        spell_sha_earthen_rage() : SpellScriptLoader("spell_sha_earthen_rage") { }
+
+        class spell_sha_earthen_rage_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_sha_earthen_rage_AuraScript);
+
+            public:
+                spell_sha_earthen_rage_AuraScript()
+                {
+                    _procTarget = nullptr;
+                }
+
+            private:
+                bool Validate(SpellInfo const* /*spellInfo*/) override
+                {
+                    if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_EARTHEN_RAGE_PASSIVE))
+                        return false;
+                    if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_EARTHEN_RAGE_PERIODIC))
+                        return false;
+                    if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_EARTHEN_RAGE_DAMAGE))
+                        return false;
+                    return true;
+                }
+
+                void HandleEffectProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+                {
+                    PreventDefaultAction();
+                    _procTarget = eventInfo.GetProcTarget();
```

Comment:
simply use  if (Unit* target = eventInfo.GetProcTarget())   and after  if (Unit* caster = eventInfo.GetActor()) so you can drop that Unit* _procTarget;  pointer.